src/Plugin/ConfigFilter/ConfigOwner.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\config_owner\Plugin\ConfigFilter;
+
+use Drupal\config_filter\Plugin\ConfigFilterBase;
+use Drupal\config_owner\OwnedConfigManagerInterface;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the filter for Config Owner.
+ *
+ * We are filtering the sync storage in view of two things:
+ *
+ * - Prevent changes to the staging (file) storage to override owned config in
+ * the active storage.
+ * - Prevent the export to the staging (file) storage of changes made to the
+ * owned config.
+ *
+ * @ConfigFilter(
+ *   id = "config_owner",
+ *   label = "Config Owner",
+ *   weight = 10,
+ * )
+ */
+class ConfigOwner extends ConfigFilterBase implements ContainerFactoryPluginInterface {
+
+  use DependencySerializationTrait;
+
+  /**
+   * Owned config manager.
+   *
+   * @var \Drupal\config_owner\OwnedConfigManagerInterface
+   */
+  protected $ownedConfigManager;
+
+  /**
+   * ConfigOwner constructor.
+   *
+   * @param array $configuration
+   *   Plugin configuration.
+   * @param string $plugin_id
+   *   Plugin ID.
+   * @param array $plugin_definition
+   *   Plugin definition.
+   * @param \Drupal\config_owner\OwnedConfigManagerInterface $ownedConfigManager
+   *   Owned config manager.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, OwnedConfigManagerInterface $ownedConfigManager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->ownedConfigManager = $ownedConfigManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('plugin.manager.owned_config')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Reading one configuration from the staging (file) storage.
+   */
+  public function filterRead($name, $data) {
+    if (!$this->isDefaultCollection()) {
+      return $data;
+    }
+
+    $owned_configs = $this->getOwnedConfig();
+    if (!isset($owned_configs[$name])) {
+      return $data;
+    }
+
+    // For owned config, we want to make sure that whatever is read from the
+    // staging storage doesn't really matter. What counts is the original owned
+    // config. This will also prevent Drupal from knowing if the staging config
+    // has changes compared to the original owned config.
+    $owned_config = $owned_configs[$name];
+    foreach ($owned_config as $key => $value) {
+      $data[$key] = $value;
```

Comment:
OK, so as discussed, we will make the following improvements:

* Third party settings should always be excluded from the ownership (sites will own these as other modules can contribute with third party settings)
* Keys should be nested like so: `colors.*` or `colors.primary`
* In case there are certain third party settings that need to be owned, they can also be specified as a key, drilling down to the place where they are found in the configuration.

The logic for merging the owned config values onto configuration should be abstracted to a single central method,